rfcs/rfc-0000-pronouns.md

```diff
@@ -0,0 +1,70 @@
+- Feature Name: Pronouns
+- Start Date: (16/06/2023)
+- RFC PR: None
+- Tracking Issue: None
+- Status: draft
+
+# Summary
+
+This change creates a pronoun field for revolt's backend.
+
+
+# Motivation
+
+Pronouns have been requested by the community and are high up on the RFC list.
+
+# Guide-level explanation
+
+Currently users display their pronouns in their Profile content (bio) but this provices a cleaner solution to the problem. The aim of this change is to ensure that pronouns for a user can be easily seen and clearly displayed seperately to a users bio.
+
+# Reference-level explanation
+
+This change will modify:
+- core/models/users
+- core/db/util/bridge
+- quark/models/users
+- core/db/models/users/ops/mongo
+  And possibly more
+
+For the backend, pronoun data will be stored as such.
+```rust
+    // User's displayed pronouns
+        #[cfg_attr(feature = "validator", validate(length(min = 1, max = 5)))]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub pronouns: Option<Vec<String>>,
+```
+
+Having pronouns on Revolt will be optional and this data will be editable from the Profile section of user settings on Revolt's main client. One of the major changes this could bring is pronoun specific language for bots interacting with users.
+
+#### Input
+```text
+?/ whois info:roles @ToastXC
+```
+#### Response
+```text
+She has no roles!
+```
+
+# Drawbacks
+
+Mocking users for displaying pronouns on their profile and creating fake pronouns as a joke is very common and seen by some as hateful, There must be a system built into the pronoun system to avoid this. That would take some time and may not be worth the effort.
```

Comment:
Pronouns should be restricted to a set database that we can easily accept contributions to, similar to Instagram.
This database shall also contain localised pronouns.

Reply:
It would also be nice if the database was consumable by downstream clients, and had information such as where the pronoun must be positioned for a language.

Reply:
discord's implementation of a fully free form text box is just *begging* to be abused. while it gives the user complete control over how their pronouns can be presented, it also means people can just, not use that space for pronouns at all. there's already dozens of examples of this textbox being filled with jokes, random quotes, fake "verified" tags, and other random things completely unrelated to pronouns.

that said, i think allowing the user to type their own pronouns is still the way to go, but the text box should simply be more restricted. i kinda like how pronouns.page does it, allowing you to type your pronouns, but requiring them to be in specific formats and only allowing a certain set of words.

Reply:
![image](https://github.com/revoltchat/rfcs/assets/38285861/87c2d6e5-3357-448d-ba2d-6de88703f2c3)

I genuinely don't know what they were thinking...

Reply:
When stated that pronouns should be localized, does that mean that the inline pronouns will be treated like UI and translated according to which language the user has on the client? Or does it mean the selection of pronouns for the user's own pronouns will be based on the user's choosen language?

Reply:
i assumed they meant it would be the former, translated based on the viewer's set language

Reply:
Ah, I see. If it is in fact the former that was planned, that makes me wonder how to address people who prefer different pronouns depending on the language being spoken. For example, there is someone who uses they/them in English, but another language they use doesn't have an equivalent they can use/like. That person speaks with someone else who has their UI language as the non-English language which tries its best to translate they/them. However, the first person either has no options for gender neutral pronouns in that language and uses certain gendered pronouns instead, or has to use a loose translation. Not sure if this is out of scope to select pronouns per language since there is no textfield for customization, but maybe this can be considered as there was a link posted earlier about how Discord's pronouns aren't thought through in non-English languages.

Reply:
> discord's implementation of a fully free form text box is just _begging_ to be abused. while it gives the user complete control over how their pronouns can be presented, it also means people can just, not use that space for pronouns at all. there's already dozens of examples of this textbox being filled with jokes, random quotes, fake "verified" tags, and other random things completely unrelated to pronouns.
> 
> that said, i think allowing the user to type their own pronouns is still the way to go, but the text box should simply be more restricted. i kinda like how pronouns.page does it, allowing you to type your pronouns, but requiring them to be in specific formats and only allowing a certain set of words.

i think revolt should go with something like how gamejolt handles pronouns (its like a list of pronouns that could be selected in order of preference):
![image](https://github.com/revoltchat/rfcs/assets/67456566/762ce0bd-a3e7-4bb0-b3c7-a86acb9e0274)